# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/mixer/src/main.rs
# Commit: eaa23cb3b396
# Full Hash: eaa23cb3b39686e114b1e78a78c2f8e5c91bcb29
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2019-12-28 04:06:02
# Regressor Bug: 1605471
# File Overlap Count: 5
# Description:
#   Bug 1605471 - Update cubeb-coreaudio to acb90e9. r=alwu
#   
#   Pick commits:
#   acb90e9 - Add license for mixer crate
#   82e7ff9 - Move criterion crate to [dev-dependencies]
# ==============================================================================

diff -r 509cd3b3ec22 -r eaa23cb3b396 third_party/rust/mixer/src/main.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/mixer/src/main.rs	Fri Dec 27 16:58:18 2019 +0000
@@ -0,0 +1,50 @@
+extern crate mixer;
+use mixer::{Channel, Mixer};
+
+fn main() {
+    // f32
+    let input_channels = vec![
+        Channel::FrontLeft,
+        Channel::Silence,
+        Channel::FrontRight,
+        Channel::FrontCenter,
+    ];
+    let output_channels = vec![Channel::FrontLeft, Channel::FrontRight];
+
+    let mut input_buffer = vec![0.0; input_channels.len()];
+    for (i, data) in input_buffer.iter_mut().enumerate() {
+        *data = (i + 1) as f32;
+    }
+    let mut output_buffer = vec![0.0; output_channels.len()];
+
+    let mixer = Mixer::new(input_channels, output_channels);
+
+    mixer.mix(&input_buffer.as_slice(), &mut output_buffer.as_mut_slice());
+    println!("{:?} is mixed to {:?}", input_buffer, output_buffer);
+
+    // i16
+    let input_channels = vec![
+        Channel::FrontLeft,
+        Channel::Silence,
+        Channel::FrontRight,
+        Channel::FrontCenter,
+        Channel::BackLeft,
+        Channel::SideRight,
+        Channel::LowFrequency,
+        Channel::SideLeft,
+        Channel::BackCenter,
+        Channel::BackRight,
+    ];
+    let output_channels = vec![Channel::Silence, Channel::FrontRight, Channel::FrontLeft];
+
+    let mut input_buffer = vec![0; input_channels.len()];
+    for (i, data) in input_buffer.iter_mut().enumerate() {
+        *data = (i + 0x7FFE) as i16;
+    }
+    let mut output_buffer = vec![0; output_channels.len()];
+
+    let mixer = Mixer::new(input_channels, output_channels);
+
+    mixer.mix(&input_buffer.as_slice(), &mut output_buffer.as_mut_slice());
+    println!("{:?} is mixed to {:?}", input_buffer, output_buffer);
+}